File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_permission_log.ts
import service from "@liujitcn/kratos-admin-core/request";
import type {
  BasePermissionLog,
  BasePermissionLogService,
  GetBasePermissionLogRequest,
  PageBasePermissionLogRequest,
  PageBasePermissionLogResponse
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_permission_log";

const BASE_PERMISSION_LOG_URL = "/v1/admin/base/permission-log";

/** Admin 权限日志服务。 */
export class BasePermissionLogServiceImpl implements BasePermissionLogService {
  /** 查询权限日志分页列表。 */
  PageBasePermissionLog(request: PageBasePermissionLogRequest): Promise<PageBasePermissionLogResponse> {
    return service<PageBasePermissionLogRequest, PageBasePermissionLogResponse>({
      url: BASE_PERMISSION_LOG_URL,
      method: "get",
      params: request
    });
  }

  /** 查询权限日志详情。 */
  GetBasePermissionLog(request: GetBasePermissionLogRequest): Promise<BasePermissionLog> {
    return service<GetBasePermissionLogRequest, BasePermissionLog>({
      url: `${BASE_PERMISSION_LOG_URL}/${request.id}`,
      method: "get"
    });
  }
}

/** 默认权限日志服务。 */
export const defBasePermissionLogService = new BasePermissionLogServiceImpl();

File: backend/internal/service/system/admin/v1/base_permission_log.go
package admin

import (
	"context"
	"fmt"

	"github.com/go-kratos/kratos/v3/log"
	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
)

// BasePermissionLogService 提供权限日志的只读管理接口。
type BasePermissionLogService struct {
	adminv1.UnimplementedBasePermissionLogServiceServer // 提供未实现 RPC 的默认返回。
	basePermissionLogCase                               *biz.BasePermissionLogCase
}

// NewBasePermissionLogService 创建权限日志管理服务。
func NewBasePermissionLogService(basePermissionLogCase *biz.BasePermissionLogCase) *BasePermissionLogService {
	return &BasePermissionLogService{basePermissionLogCase: basePermissionLogCase}
}

// PageBasePermissionLog 查询权限日志分页列表。
func (s *BasePermissionLogService) PageBasePermissionLog(ctx context.Context, req *adminv1.PageBasePermissionLogRequest) (*adminv1.PageBasePermissionLogResponse, error) {
	page, err := s.basePermissionLogCase.PageBasePermissionLog(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBasePermissionLog %v", err))
		return nil, errorsx.WrapInternal(err, "查询权限日志分页列表失败")
	}
	return page, nil
}

// GetBasePermissionLog 查询权限日志详情。
func (s *BasePermissionLogService) GetBasePermissionLog(ctx context.Context, req *adminv1.GetBasePermissionLogRequest) (*adminv1.BasePermissionLog, error) {
	item, err := s.basePermissionLogCase.GetBasePermissionLog(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBasePermissionLog %v", err))
		return nil, errorsx.WrapInternal(err, "查询权限日志详情失败")
	}
	return item, nil
}
